addons/website_project_timesheet/security/portal_security.xml

```diff
@@ -0,0 +1,18 @@
+<?xml version="1.0" encoding="utf-8"?>
+<openerp>
+    <data noupdate="1">
+        <record id="portal_task_timesheet_rule" model="ir.rule">
+            <field name="name">Project/Task Timesheets: frontend access</field>
+            <field name="model_id" ref="analytic.model_account_analytic_line"/>
+            <field name="domain_force">['|',
+                                            '&amp;',
+                                                ('task_id.project_id.privacy_visibility', '=', 'portal'),
+                                                ('task_id.project_id.message_partner_ids', 'child_of', [user.partner_id.commercial_partner_id.id]),
+                                            '&amp;',
```

Comment:
('task_id.project_id.privacy_visibility', '=', 'portal'),
('task_id.message_partner_ids', 'in' , [user.partner_id.id]) COMPANY